src/main/java/org/fcrepo/oai/generator/JcrOaiEtdmsGenerator.java

```diff
@@ -214,8 +214,22 @@ public Thesis generate(final Session session, final Container obj, final String
                 break;
 
             case "dcterms:abstract":
+               /*
+               * In the context of the Fedora 4.2 & HydraNorth stack,
+               * if dcterms:description property (in jcr/xml) is of a
+               * substantial length (tested with 6760 characters) the property
+               * type switches to `Binary` with subsequent edits appending a new
+               * value to the list of values for the property (i.e., not replacing)
+               * This workaround chooses the last value assuming the previous
+               * values are old versions (to avoid returning all values of the
+               * property, including the obsolete). 2017-05-12
+               */
+                Value lastValue = null;
                 for (final Value v : prop.getValues()) {
```

Comment:
@piyapongch 

prop.getLength() returns the length of the value (not the length of the array of values). Trying another way.

```
final Value[] vals = prop.getValues();
final int len = java.lang.Math.toIntExact(vals.length);
if (len > 0) {
    final Value lastValue = (len > 0) ? vals[len - 1] : null;
    addDescription(lastValue, thesis.getDescription(), prefix);
    }
```